features/mbedtls/targets/TARGET_Cypress/TARGET_PSOC6/ecdsa_alt.c

```diff
@@ -0,0 +1,252 @@
+    ECDSA_VALIDATE_RET( s     != NULL );
+    ECDSA_VALIDATE_RET( d     != NULL );
+    ECDSA_VALIDATE_RET( f_rng != NULL );
+    ECDSA_VALIDATE_RET( buf   != NULL || blen == 0 );
+
+    /* Fail cleanly on curves such as Curve25519 that can't be used for ECDSA */
+    if( grp->N.p == NULL )
+        return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
+
+    key.curveID = cy_get_dp_idx(grp->id);
+    ECDSA_VALIDATE_RET( key.curveID != CY_CRYPTO_ECC_ECP_NONE);
+
+    /* Make sure d is in range 1..n-1 */
+    if( mbedtls_mpi_cmp_int( d, 1 ) < 0 || mbedtls_mpi_cmp_mpi( d, &grp->N ) >= 0 )
+        return( MBEDTLS_ERR_ECP_INVALID_KEY );
+
+    /* Reserve the crypto hardware for the operation */
+    cy_hw_crypto_reserve(&crypto_obj, CYHAL_CRYPTO_VU);
+
+	dp = Cy_Crypto_Core_ECC_GetCurveParams(key.curveID);
+
+	bytesize = CY_CRYPTO_BYTE_SIZE_OF_BITS(dp->size);
+
+	key.k = malloc(bytesize);
+    MBEDTLS_MPI_CHK((key.k == NULL) ? MBEDTLS_ERR_ECP_ALLOC_FAILED : 0);
+
+    MBEDTLS_MPI_CHK( mbedtls_mpi_write_binary( d, key.k, bytesize ) );
+    Cy_Crypto_Core_InvertEndianness(key.k, bytesize);
+
+    sig = malloc(2 * bytesize);
+    MBEDTLS_MPI_CHK((sig == NULL) ? MBEDTLS_ERR_ECP_ALLOC_FAILED : 0);
+
+    tmp_k = malloc(bytesize);
+    MBEDTLS_MPI_CHK((tmp_k == NULL) ? MBEDTLS_ERR_ECP_ALLOC_FAILED : 0);
+
+    ecdsa_status = Cy_Crypto_Core_ECC_MakePrivateKey(crypto_obj.base, key.curveID, tmp_k, f_rng, p_rng);
+    MBEDTLS_MPI_CHK((ecdsa_status == CY_CRYPTO_SUCCESS) ? 0 : MBEDTLS_ERR_ECP_HW_ACCEL_FAILED);
+
+    ecdsa_status = Cy_Crypto_Core_ECC_SignHash(crypto_obj.base, buf, blen, sig, &key, tmp_k);
+    MBEDTLS_MPI_CHK((ecdsa_status == CY_CRYPTO_SUCCESS) ? 0 : MBEDTLS_ERR_ECP_HW_ACCEL_FAILED);
+
+    /* Prepare a signature to load into an mpi format */
+    Cy_Crypto_Core_InvertEndianness(sig, bytesize);
+    Cy_Crypto_Core_InvertEndianness(sig + bytesize, bytesize);
+
+    MBEDTLS_MPI_CHK( mbedtls_mpi_read_binary( r, sig, bytesize ) );
+    MBEDTLS_MPI_CHK( mbedtls_mpi_read_binary( s, sig + bytesize, bytesize ) );
+
+cleanup:
+    /* Realease the crypto hardware */
+    cy_hw_crypto_release(&crypto_obj);
+
+    if (key.k != NULL)
+    {
+        mbedtls_platform_zeroize(key.k, bytesize);
+    	free(key.k);
+    }
+    if (sig != NULL)
+    {
+        mbedtls_platform_zeroize(sig, 2 * bytesize);
+    	free(sig);
+    }
+    if (tmp_k != NULL)
+    {
+        mbedtls_platform_zeroize(tmp_k, bytesize);
+        free(tmp_k);
+    }
+
+    return( ret );
+}
+#endif /* MBEDTLS_ECDSA_SIGN_ALT */
+
+#if defined(MBEDTLS_ECDSA_VERIFY_ALT)
+/*
+ * Verify ECDSA signature of hashed message (SEC1 4.1.4)
+ * Obviously, compared to SEC1 4.1.3, we skip step 2 (hash message)
+ */
+int mbedtls_ecdsa_verify( mbedtls_ecp_group *grp,
+                  const unsigned char *buf, size_t blen,
+                  const mbedtls_ecp_point *Q, const mbedtls_mpi *r, const mbedtls_mpi *s)
+{
+    int ret;
+	uint8_t stat;
+    size_t bytesize;
+	size_t olen;
```

Comment:
are here tabs, line 179 is also unaligned

Reply:
Fixed in the commit 08fd157